plugins/snowflake/credentials.go

```diff
@@ -0,0 +1,108 @@
+package snowflake
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func Credentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.Credentials,
+		DocsURL: sdk.URL("https://docs.snowflake.com/en/user-guide/snowsql.html"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Account,
+				MarkdownDescription: "Snowflake account name.",
+				Optional:            false,
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Snowflake username.",
+				Optional:            false,
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to Snowflake account.",
+				Optional:            false,
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TrySnowflakeConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"SNOWSQL_ACCOUNT": fieldname.Account,
+	"SNOWSQL_USER":    fieldname.Username,
+	"SNOWSQL_PWD":     fieldname.Password,
+}
+
+func TrySnowflakeConfigFile() sdk.Importer {
+	return importer.TryFile("~/.snowsql/config", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		credentialsFile, err := contents.ToINI()
+		if err != nil {
+			out.AddError(err)
+			return
+		}
+
+		// The generated ~/.snowsql/config file, by default, includes an uncommented [connections.example] credential section below the actual
+		// [connections] section. We save the accountname, username, and password listed under [connections] then stop parsing the file
+		fields := make(map[sdk.FieldName]string)
+		for _, section := range credentialsFile.Sections() {
```

Comment:
I think we should rework this part because, for initial setup (that is, a config file without anything under `[connections]`) results in `[connections.example]`'s values getting added to 1Password:

<img width="682" alt="image" src="https://user-images.githubusercontent.com/18581859/214242281-9fc669a0-7340-47a2-9f77-12531594fba3.png">

We should prompt the user to enter the credentials manually instead.

Something like this should do the trick:

```suggestion
		for _, section := range credentialsFile.Sections() {
			if !section.HasKey("accountname") && !section.HasKey("username") && !section.HasKey("password") {
				return
			}
```

Reply:
Likewise, the parts below could be wrapped with a `section.Name() == "[connections]"`, I think. I haven't really tested that though but can give it a shot tomorrow morning:

```
if section.HasKey("accountname") && section.Key("accountname").Value() != "" {
	fields[fieldname.Account] = section.Key("accountname").Value()
}

if section.HasKey("username") && section.Key("username").Value() != "" {
	fields[fieldname.Username] = section.Key("username").Value()
}

if section.HasKey("password") && section.Key("password").Value() != "" {
	fields[fieldname.Password] = section.Key("password").Value()
}
```

Reply:
Great find, Arun! Thank you

Reply:
Thanks for the suggestion Arun! Wrapping the conditionals with `if section.Name() == "connections"` as you suggested worked great and the importer behaviour works as expected. The `if !section.HasKey("accountname") && !section.HasKey("username") && !section.HasKey("password")` checks seem to be already covered by the conditionals I wrote so I don't think it's necessary to cover those again.

Reply:
Jup, with your updated changes, this looks right to me!